### ISSUE TITLE
MacOS Permission problem in VSCode 

### ISSUE THREAD

**Issue by berkobob** [flutter/flutter#114352]
I am writing a simple [plugin](https://github.com/berkobob/reminders) to read, write, edit & delete Apple Reminders. The first thing I do is request access to reminders:

```swift
eventStore.requestAccess(to: EKEntityType.reminder, completion: {(granted, error) in 
            if let error = error { print(error) }
            self.hasAccess = granted ? true : false
            })
```

When I run my app from the terminal (iTerm) using `flutter run -d macOS` a window popped up to allow me to grant access to my Reminders. If I build the app and run the executable it also works. When I run my app from within VSCode, either using Debug or the VSCode terminal, the permission window does not appear and my app fails to get permission. I'm thinking this is a VSCode issue as it works everywhere else.

I have the same problem on my M1 MacStudio and my Intel MacBook Pro. 

How do I grant access to my Reminders to VSCode?

<details>
<summary>flutter doctor -v</summary>
[✓] Flutter (Channel stable, 3.3.6, on macOS 13.0 22A380 darwin-arm, locale en-GB)
    • Flutter version 3.3.6 on channel stable at /Users/antoine/library/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 6928314d50 (6 days ago), 2022-10-25 16:34:41 -0400
    • Engine revision 3ad69d7be3
    • Dart version 2.18.2
    • DevTools version 2.15.0

[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.0)
    • Android SDK at /Users/antoine/Library/Android/sdk
    • Platform android-33, build-tools 33.0.0
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 14.0.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14A400
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2021.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)

[✓] VS Code (version 1.72.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.50.0

[✓] Connected device (2 available)
    • macOS (desktop) • macos  • darwin-arm64   • macOS 13.0 22A380 darwin-arm
    • Chrome (web)    • chrome • web-javascript • Google Chrome 107.0.5304.87

[✓] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!
</details>

![VSCode Version](https://user-images.githubusercontent.com/6568107/199027305-764ffe49-d8ac-4af2-9697-23d2c510b3a5.png)

**Comment by exaby73** [comment]
Hello @berkobob. Thank you for filing this issue. Can you please provide the following information?

- A [complete and minimal, reproducible example](https://stackoverflow.com/help/minimal-reproducible-example). Ideally, a single `main.dart` file that can be run directly, or a public Github repo.
- The output of `flutter run-v`

Please provide the information in the form of text. See how to make collapsible sections with Markdown [here](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/organizing-information-with-collapsed-sections).

**Comment by berkobob** [comment]
Here is a public repo. I've included the absolute minimum necessary to reproduce the error but as this is a plugin, there is more than `main.dart`. `main.dart` is in the `example/lib folder`: https://github.com/berkobob/example

The output from `flutter run -v` is over 2.5K lines long so I'm attaching a text file instead of collapsible text. I hope that's OK. Let me know if you still need me to cut and paste:
 
[run-v.txt](https://github.com/flutter/flutter/files/9919702/run-v.txt)

Screenshot when run from VSCode:
![Screenshot when run from VSCode](https://user-images.githubusercontent.com/6568107/199474096-7a8792ee-e763-4e4e-a728-640766a980f5.png)

And this is the exact same code being run from the terminal (iTerm2):
![Screenshot when run from terminal](https://user-images.githubusercontent.com/6568107/199474602-a1cf75fc-bde7-4fce-894d-e7e0675b83f4.png)

I don't know if you need this but here it the output from 'flutter run -v' when run from the terminal:
[run-v-from-terminal.txt](https://github.com/flutter/flutter/files/9919746/run-v-from-terminal.txt)

**Comment by exaby73** [comment]
I cannot reproduce this issue. I can see `true` when running the app from VSCode

**Comment by exaby73** [comment]
Yes Visual Studio Code is showing up in this list, with it switched on. Looks like this issue is with VSCode not with Flutter directly.

Please see https://flutter.dev/community for resources and asking questions like this,
you may also get some help if you post it on Stack Overflow and if you need help with your code, please see https://www.reddit.com/r/flutterhelp/. Alternatively, you can raise an issue in the [vscode repo](https://github.com/microsoft/vscode/issues)
Closing, as this isn't an issue with Flutter itself. If you disagree, please write in the comments and I will reopen it.
Thank you

**Comment by berkobob** [comment]
Just before you close this issue, I think this repo is the right place to log issues with the Flutter extension for VSCode team. Is it possible for them to take a quick look please. Thanks.

**Comment by DanTup** [comment]
I was able to reproduce this on my M1 MacBook. I was able to force the permission prompt by pasting this into the built-in VS Code terminal:

```
osascript - title <<END
on run a
tell app "Reminders"
tell list "Reminders" of default account
make new reminder with properties {name:item 1 of a}
end
end
end
END
```

(I got this from https://apple.stackexchange.com/questions/66981/how-can-i-add-reminders-via-the-command-line).

That triggered the prompt and after accepting, running the app now shows it has permission. I'm not sure why the code in the app isn't triggering the prompt with the call to `EKEventStore().requestAccess()` though.

**Comment by berkobob** [comment]
Thanks so much Dan. Much appreciated! Kudos.

**Comment by exaby73** [comment]
@DanTup Just to confirm, is this a Vscode issue or a Flutter issue?

**Comment by DanTup** [comment]
@exaby73 I don't know a lot about it, but it feels to me like the issue is on macOS or in the SDK used by the swift code in the app. The code was clearly asking for access to Reminders but macOS was never showing a prompt, nor adding the app to the Reminders section in the Security&Privacy section. For some reason, running a script that tries to access reminders from the terminal _does_ trigger the prompt, allowing permission to be granted.

I don't think it's a VS Code (or Dart-Code) issue. I suspect it's not a Flutter issue, but I couldn't be sure. It's possible something about how Flutter is building the macOS app is preventing that permissions prompt (or something specific about that example app), but I don't know much about how that works. I just did some searching and found [this similar issue](https://stackoverflow.com/questions/41469283/ekeventstore-requestaccess-is-failing-to-display-the-user-prompt-in-ios-9) that was caused by not having a `CFBundleDisplayName` name set, so maybe it's something similar to that? 🤔

**Comment by exaby73** [comment]
I'll label this issue for further investigation and insights. Thank you :)

**Comment by kingvhit** [comment]
Has same issue when running on Android Studio. 

Just run by command-line or dirrectly from xcode will work fine. Otherwise, run from Android Studio - Run button has no effect (no permission request)

**Comment by muknta** [comment]
I have the same issue.
MacBook Pro 2021 16"
Macos Sonoma 14.3.1

Macos builds from Android Studio and console are not showing the permission dialog on my machine.
Macos build from Xcode 15 works fine every time.

**Comment by muknta** [comment]
solved locally:
Issue was observed in non-sandbox Macos build. Entitlements file with permissions was not recognized using custom build notarizing.
So issue solves if we either move to Macos sandbox, or pass the entitlements to custom notirizing command in CI/CD

**Comment by IVLIVS-III** [comment]
During development of the [device_calendar](https://github.com/builttoroam/device_calendar) plugin we ran into the same issue with macOS development. When running a plugin-consuming flutter app (e.g. the plugin's example app) from "Terminal.app" (`flutter run -d macos`) permissions worked fine.
However, the same command when run in the integrated IDE terminal (Android Studio / VS Code) results in neither permissions nor a permissions prompt.

The issue boils down to the fact that the IDE itself and thus the integrated terminal and by extension also the flutter app launched via the integrated terminal all share the IDE permissions. However the permissions request from the flutter app does not change the IDE permissions.

A solution is to request and grant permissions for the IDE once. Afterwards launching the flutter app from within the IDE (integrated terminal or launch UI button) works with permissions set properly.
I've attached the original message from a discussion on our plugin's issues that includes a shell command which can be pasted into the integrated terminal to request and grant calendar permissions.
It should be adaptable to other APIs.

---

I've created a shell command that can be run once from within the integrated terminal in Android Studio / VS Code / another IDE.
Afterwards, launching the macOS app from within the IDE should work with permissions set.

```bash
tccutil reset Calendar $(mdls -name kMDItemCFBundleIdentifier "$(mdfind "kMDItemFSName == '$(ps -o comm= -p $(ps -o ppid= -p $$) | awk -F'.app' '{print $1".app"}' | awk -F'/' '{print $NF}')'" | head -n 1)" | sed 's/.*= "\(.*\)"/\1/') && osascript -e 'tell application "Calendar" to get the name of every calendar'
```

<details>
<summary>Command explanation (click me)</summary>

In the following explanation `<<i>>` is a placeholder for the command from step `i` verbatim:

1. `ps -o comm= -p $(ps -o ppid= -p $$)`
    Detects which executable hosts the integrated terminal, e.g. `/Applications/Android Studio.app/Contents/MacOS/studio`

2. `<<1>> | awk -F'.app' '{print $1".app"}' | awk -F'/' '{print $NF}'`
    Extracts the application name (IDE), e.g `Android Studio.app`

3. `mdfind "kMDItemFSName == '$(<<2>>)'" | head -n 1`
    Finds the application path, e.g. `/Applications/Android Studio.app`

*Note: For VS Code the host executable path does not start with the application path, hence we need steps 2 and 3 instead of just truncating the output from step 1.*

4. `mdls -name kMDItemCFBundleIdentifier "$(<<3>>)"`
    Finds the application bundle identifier, e.g. `kMDItemCFBundleIdentifier = "com.google.android.studio"`

5. `<<4>> | sed 's/.*= "\(.*\)"/\1/'`
    Extracts the application bundle identifier, e.g. `com.google.android.studio`

6. `tccutil reset Calendar $(mdls -name kMDItemCFBundleIdentifier "$(<<5>>)`
    Resets the calendar permissions for the IDE

7. `<<6>> && osascript -e 'tell application "Calendar" to get the name of every calendar'`
    Forces a calendar permission prompt by trying to list all calendar names
</details>

**Comment by jakobwarnhjelm** [comment]
I've researched this for a flutter app requesting permissions to view photo library.
Took quite some time, I thought erroneously  that Xcode did something special because it worked there from the start.

If I use flutter cli and do `flutter run -d macos` the MacOS permissions dialog doesn't appear.
However, I can go to `../build/macos/Build/Products/Debug` and launch it manually. Then it will work.

**In summary: when flutter CLI launches the app, permissions doesn't work. When I launch it, it does work.**

Also note that builds can appear in `/Users/<name>/Library/Developer/Xcode/DerivedData` depending on how you build.

So my workflow will need to be something like:

>  `flutter build macos --debug`
> `cd  ../build/macos/Build/Products/Debug`
> `open <appname>.app`

**Comment by lukemmtt** [comment]
@jakobwarnhjelm Thats good to know as another option of a workaround, although it's worth noting that the command in Julius's previous comment above (tccutil etc) should be able to not only work around the issue, but preserve the ability to use the **flutter debugger** as well—it did for me, which was massively helpful in my development efforts. 

Before I found Julius’s solution, I was completely unable to debug any permission-required features in my app (Calendar/Reminders in my case). I had to run the app, close it, then restart it from Xcode directly without the Flutter debugger. Julius’s command completely fixed this; it forces macOS to show the permission prompt for your IDE or terminal, and once granted, you can `flutter run` normally with permissions working AND the debugger staying connected, not just for one session but for all future sessions.

Specifically, for your use case, I would expect that you just need to replace "Calendar" with the permission of interest: either "Photos" or "PhotosAdd". 

Perhaps you tried an approach like this and it didn't work? Or perhaps you're content with the manual workaround, all good. I'm happy to help troubleshoot if you're interested in getting Julius's approach working though.
